src/memo/memo.ml

```diff
@@ -797,41 +731,26 @@ let get_cached_value_in_current_cycle (dep_node : _ Dep_node.t) =
 module Cached_value = struct
   include M.Cached_value
 
-  let capture_dep_values ~deps_rev =
```

Comment:
I hesitated when deleting this function. It's no longer used, but it was doing some useful sanity checks, so we could keep it in some form, perhaps, by moving these checks to more appropriate places.

Reply:
Indeed, historically the checks in this function have caught quite a lot of bugs.
I think having the checks could be useful, but I think they'd have to be guarded by a debug flag, since removing this traversal of the deps is probably the main contribution of this optimization.

Reply:
Agreed, I'll put it behind a debug flag, enabled in the testsuite.

Reply:
Done.